app/src/main/java/com/portfolio/tasky/entry/fragments/RegistrationFragment.kt

```diff
@@ -64,6 +76,17 @@ class RegistrationFragment : Fragment(), FragmentInflater by FragmentInflaterImp
         viewModel.validateFields.observe(viewLifecycleOwner) {
             it.let {  viewBinding.btnReg.isEnabled = it == true }
         }
+
+        viewModel.networkObserver.observe(viewLifecycleOwner){
```

Comment:
No need to have a network observer on the login/registration

You could just make the API call, and display the error

Reply:
Instead of showing error from api call, I disable the buttons when network is not available.